### PR TITLE
Beta fixes for Muon Analysis: keep selections on tab change

### PR DESCRIPTION
**Pull request by tom-perkins**
Fix points raised by beta testers:
- Group selection forgotten on switching between tabs
- Fit label should be auto-set to run number
- Remove unnecessary warning about labels on sequential fit

Unit tests have been added to cover these changes.

**To test:**
_Data sets: `MUSR00015189-90` from unit test data_
- Set facility to ISIS, open _Interfaces/Muon/Muon Analysis_ and load `MUSR00015190.nxs`.
- Go to "Data Analysis" tab. 
  - The simultaneous fit label (centre of page) should be the run number, greyed out.
  - Tick checkboxes for groups `fwd` and `long` (label will be enabled)
- Switch to another tab, then back to "Data Analysis"
  - Verify the groups `fwd` and `long` are still selected
- Go to "Home" tab and load `MUSR00015189.nxs`
- Return to "Data Analysis" tab
  - The fit label should have changed to "15189" to match the run number
  - The group selection should be the same (`fwd` and `long`)
  - Change the fit label to "hello"
- Switch to "Home" tab and load `MUSR00015190.nxs` again
- Return to "Data Analysis" tab
  - This time the fit label should not have changed (as it won't replace a user-supplied one)

Re #17602 (does not close that issue as it contains more points)

No release notes. The new fitting UI is already in the release notes for 3.8.

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
- [x] Is the code of an acceptable quality?
- [x] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)? Is it well structured with small focussed classes/methods/functions?
- [x] Are there unit/system tests in place? Are the unit tests small and test the a class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?
##### Functional Tests
- [x] Do changes function as described? Add comments below that describe the tests performed?
- [ ] How do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant documentation been added/updated?
- [ ] Is user-facing documentation written in a user-friendly manner?
- [ ] Has developer documentation been updated if required?
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by KarlPalmen** [comment]
When testing, I found (as expected) one first has to add a function before any of the fit options were available. Whenever the fit options were available sequential fit was available, even if only one group is selected. This is not what the test as described requires.

File: MantidQt/CustomInterfaces/test/MuonAnalysisFitDataPresenterTest.h [hunk 1]
```diff
@@ -148,6 +148,7 @@ class MuonAnalysisFitDataPresenterTest : public CxxTest::TestSuite {
   }
 
   void test_setAssignedFirstRun_singleWorkspace() {
+    setupGroupPeriodSelections();
```
Comment (KarlPalmen) [inline MantidQt/CustomInterfaces/test/MuonAnalysisFitDataPresenterTest.h]: Should this not be called something like `doTestSetupGroupPeriodSeclections`?

Comment (tom-perkins) [inline MantidQt/CustomInterfaces/test/MuonAnalysisFitDataPresenterTest.h]: This just sets up what some methods will return when called, it doesn't test anything itself

Comment (KarlPalmen) [inline MantidQt/CustomInterfaces/test/MuonAnalysisFitDataPresenterTest.h]: Oops! I misread the code and thought an early call of `setupGroupPeriodSelections` in `test_setSelectedWorkspace` was the name of that latter function, having expected you to have added code for `setupGroupPeriodSelections`.

**Comment by KarlPalmen** [comment]
I think I may have misunderstood 

> ◦The simultaneous fit label (centre of page) should be the run number, greyed out.

I thought it referred to sequential fit in the fit menu, revealed when one presses the fit button.

**Comment by KarlPalmen** [comment]
Now I see you are referring to the label. I now see that is greyed out whenever one group is selected and not whenever more than one group is selected. This I now take to be what is meant by the test instructions (not `sequential fit' in the middle of the fit menu).

**Comment by KarlPalmen** [comment]
:shipit:

**Review by martyngigg — APPROVED** [review]

**Pull request merged**